File: run_length_encoding/run_length_encoding.py
def run_length_encoding(string):
  current_run_length = 1
  encoded_string_characteres = []
  for i in range(1, len(string)):
    previous_character = string[i - 1]
    current_character = string[i]
    if current_character != previous_character or current_run_length == 9:
      encoded_string_characteres.append(str(current_run_length))
      encoded_string_characteres.append(previous_character)
      current_run_length = 0
    current_run_length += 1
  encoded_string_characteres.append(str(current_run_length))
  encoded_string_characteres.append(string[len(string) - 1])
  return "".join(encoded_string_characteres)


print(run_length_encoding("aaaaaaaaaaaaaaaaabbbbbbbbbbbbbccccccccccc"))

File: run_length_encoding/run_length_encoding.go
package main

import (
	"fmt"
	"strings"
)

func runLengthEncoding(s string) string {
	currentRunLength := 1
	encodedStringCharacters := []string{}
	for i := 1; i < len(s); i++ {
		previousCharacter := s[i-1]
		currentCharacter := s[i]
		if currentCharacter != previousCharacter || currentRunLength == 9 {
			encodedStringCharacters = append(encodedStringCharacters, fmt.Sprintf("%d", currentRunLength))
			encodedStringCharacters = append(encodedStringCharacters, string(previousCharacter))
			currentRunLength = 0
		}
		currentRunLength++
	}

	encodedStringCharacters = append(encodedStringCharacters, fmt.Sprintf("%d", currentRunLength))
	encodedStringCharacters = append(encodedStringCharacters, string(s[len(s)-1]))
	return strings.Join(encodedStringCharacters, "")
}

func main() {
	input := "aaaaaaaaaaaaaaaaabbbbbbbbbbbbbccccccccccc"
	encoded := runLengthEncoding(input)
	fmt.Println(encoded)
}
